pkg/kubectl/cmd/util/helpers.go

```diff
@@ -400,3 +401,21 @@ func UpdateObject(info *resource.Info, updateFn func(runtime.Object) error) (run
 	}
 	return info.Object, nil
 }
+
+type groupID int
+
+const (
+	unknownGroup groupID = iota
+	apiGroup
+	experimentalGroup
+)
+
+func groupForResource(resource string) groupID {
```

Comment:
The rest mapper can infer a group using logic that would eventually come from the api version.  If I as a client say "I want to use object Foo" and Foo is not adequately scoped, a group has to be chosen.  We aren't going to break every object already serialized, and we have two API groups today (Kube, and OpenShift).  Resource names and kinds either have to be unique across groups or disambiguated by callers (via preference or precedence or interaction).  So RESTMapper is going to have to do this disambiguation anyway.